### PR TITLE
Add several methods

### PR DESCRIPTION
**Pull request by aryzing**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,11 +25,13 @@
...
     "@types/bun": "latest",
     "prettier": "^3.3.3",
-    "tsup": "^8.2.4"
+    "tsup": "^8.3.5"
   },
   "peerDependencies": {
     "typescript": "^5.0.0",
-    "valibot": "^0.41.0"
+    "@stacks/blockchain-api-client": "^8.2.1",
+    "@stacks/transactions": "^7.0.0",
```
Comment (fedeerbes) [inline package.json:33]: question: what is best here? adding a min version or directly putting the major version we want also typescript should be devDependencies ?

**Review by fedeerbes — DISMISSED** [review]
LGTM, left some questions. =)

File: package.json (hunk 1 above)
Comment (aryzing) [reply package.json:33]: Unsure what's best. I believe the ideal would be the lowest possible compatible version, although not sure it'd be cost-effective to figure that out. I think for now we're ok if we bump the versions to what we need until we find a reason not to.

And well spotted on TS! I'll move it to devdeps

File: src/stacks-api/transactions/mempool-transactions.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  );
+
+  if (!res.ok) {
+    return error({
+      name: "FetchMempoolTransactionsError",
+      message: "Failed to fetch mempool transactions.",
+      data: {
+        status: res.status,
+        statusText: res.statusText,
+        bodyParseResult: await safePromise(res.json()),
```
[the remark this answers is not in the document]
Comment (aryzing) [reply src/stacks-api/transactions/mempool-transactions.ts:86]: Yes. It's part of the error payload data, so it's fine. However, you just reminded me that not all responses return json, and it's safer to go with `res.text()` so I'll use that instead

**Review by fedeerbes — DISMISSED** [review]

File: src/README.md
Comment (fedeerbes) [inline src/README.md:1]: 🙌🏼

**Pull request merged**